src/lib/y2storage/device.rb

```diff
@@ -36,15 +36,37 @@ class Device
     wrap_class Storage::Device,
       downcast_to: ["BlkDevice", "Mountable", "PartitionTables::Base", "LvmPv", "LvmVg"]
 
-    # @!method ==(device)
-    #   Compare two devices.
-    #   @note Devices are equal if they have the same {#sid storage id}.
-    #   @see sid
+    storage_forward :storage_eql, to: :==
+    protected :storage_eql
+
+    #  compare two devices.
+    #  @note devices are equal if they have the same {#sid storage id}.
+    #  @see sid
+    #
+    #  @param other [Device]
+    #  @return [Boolean] false if compared to different class
+    def ==(other)
+      return false if self.class != other.class
+
+      storage_eql(other)
+    end
+
+    #  compare two devices.
+    #  @see ==
     #
-    #   @param device [Device]
-    #   @return [Boolean]
-    storage_forward :==
-    storage_forward :!=
+    #  @param other [Device]
+    #  @return [Boolean] true if compared to different class
+    def !=(other)
+      !(self == other)
+    end
+
+    alias_method :eql?, :==
+
+    # redefine hash method to return same result for same devices indenpendently
```

Comment:
typo: independently

On the other hand, since this definition of `#hash` is kind of unexpected and somebody could feel tempted to replace it with something more colorful, I would appreciate if you mention here that this is done in order to make the `Array` operations work. Maybe even mentioning the bug number.

Reply:
ok